homeassistant/components/switch/broadlink.py

```diff
@@ -268,3 +279,56 @@ def _update(self, retry=2):
         if state is None and retry > 0:
             return self._update(retry-1)
         self._state = state
+
+
+class BroadlinkMP1Switch(BroadlinkRMSwitch):
+    """Representation of an Broadlink switch."""
+
+    def __init__(self, friendly_name, device, slot):
+        """Initialize the switch."""
+        super().__init__(friendly_name, device, None, None)
+        self._command_on = 1
+        self._command_off = 0
+        self._slot = slot
+ 
+    @property
+    def assumed_state(self):
+        """Return true if unable to access real state of entity."""
+        return False
+      
+    def _sendpacket(self, packet, retry=2):
+        """Send packet to device."""
+        try:
+            self._device.set_power(self._slot, packet)
+        except (socket.timeout, ValueError) as error:
+            if retry < 1:
+                _LOGGER.error(error)
+                return False
+            if not self._auth():
+                return False
+            return self._sendpacket(packet, max(0, retry-1))
+        return True
+
+    @property
+    def should_poll(self):
+        """Polling needed."""
+        return True
+
+    def update(self):
+        """Synchronize state with switch."""
+        self._update()
+
+    def _update(self, retry=2):
+        try:
+            states = self._device.check_power()
```

Comment:
I think we here will ask for the same data 4 times (once for each slot).
You should make a data object that you can pass to the init function instead of the device object. The data object should handle the check_power call.
Here is an example: https://github.com/home-assistant/home-assistant/blob/38071501b421627dadbceb4bbdced195ac4ec620/homeassistant/components/switch/digitalloggers.py#L137

Reply:
updated